synapse/rest/client/v1/login.py

```diff
@@ -67,6 +77,16 @@ def on_POST(self, request):
                     "uri": "%s%s" % (self.idp_redirect_url, relay_state)
                 }
                 defer.returnValue((200, result))
+            elif self.cas_enabled and (login_submission["type"] ==
+                                       LoginRestServlet.CAS_TYPE):
+                url = "%s/proxyValidate" % (self.cas_server_url)
+                parameters = {
+                    "ticket": login_submission["ticket"],
+                    "service": login_submission["service"]
+                }
+                response = requests.get(url, verify=False, params=parameters)
```

Comment:
You really don't want to be doing blocking requests in the event loop. There's a `synapse.http..SimpleHttpClient` that probably does what you want.